### PR TITLE
feat: event logs clean

### LINKED ISSUE #255: event logs clean
**What's it about?**

**What's the reason why we need it?**

I believe this is an important feature for KubeClipper. There're a few use cases:

* case one
* case two
* ...

Please leave your comments below if there's anyone agrees with me. Or just give me a thumb up.

**Area Suggestion**

/kind feature-request

### PR DESCRIPTION
**Pull request by Xvv-v**
Signed-off-by: xu.jingwen <xu.jingwen@99cloud.net>

### What type of PR is this?
/kind feature

### Which issue(s) this PR fixes:

Fixes #255 

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/auditing/option/options.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/apis/audit"
+)
+
+var (
+	allowedlevel = sets.NewString(string(audit.LevelNone), string(audit.LevelMetadata), string(audit.LevelRequest), string(audit.LevelRequestResponse))
+)
+
+type AuditOptions struct {
+	EventLogHistoryRetentionPeriod time.Duration `json:"eventLogHistoryRetentionPeriod" yaml:"eventLogHistoryRetentionPeriod"`
```
Comment (x893675) [inline pkg/auditing/option/options.go:19]: rename RetentionPeriod
rename MaximumEntries

File: pkg/controller/audit_status.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+func (s *AuditStatusMon) SetupWithManager(mgr manager.Manager) {
+	s.mgr = mgr
+	s.logger = mgr.GetLogger().WithName("audit-status-monitor")
+	mgr.AddWorkerLoop(s.monitorAuditStatus, auditStatusMonitorPeriod)
+}
+
+func (s *AuditStatusMon) monitorAuditStatus() {
+	timestamp := time.Now().Add(-s.AuditOptions.EventLogHistoryRetentionPeriod).Format(time.RFC3339)
+	s.cleanLogs("type", timestamp)
```
Comment (lixd) [inline pkg/controller/audit_status.go:34]: wrong FieldSelector format

Comment (x893675) [inline pkg/controller/audit_status.go:34]: add annotate
// clean login / logout audit envent
s.cleanLogs("type", timestamp)

File: pkg/server/config/config.go [hunk 3]
```diff
@@ -56,6 +58,7 @@ type Config struct {
 	MQOptions               *natsio.NatsOptions                `json:"mq,omitempty" yaml:"mq,omitempty"  mapstructure:"mq"`
 	LogOptions              *logger.Options                    `json:"log,omitempty" yaml:"log,omitempty" mapstructure:"log"`
 	AuthenticationOptions   *authoptions.AuthenticationOptions `json:"authentication,omitempty" yaml:"authentication,omitempty" mapstructure:"authentication"`
+	AuditOptions            *auditoptions.AuditOptions         `json:"auditOptions,omitempty" yaml:"auditOptions,omitempty" mapstructure:"authentication"`
```
Comment (lixd) [inline pkg/server/config/config.go:61]: audit tag not same with kubeclipper-server.yaml

File: cmd/kubeclipper-server/app/options/options.go [hunk 4]
```diff
@@ -54,16 +54,21 @@ func (s *ServerOptions) Flags() (fss cliflag.NamedFlagSets) {
...
 
 func (s *ServerOptions) Validate() []error {
+	var err error
 	var errors []error
 	errors = append(errors, s.GenericServerRunOptions.Validate()...)
 	errors = append(errors, s.EtcdOptions.Validate()...)
 	errors = append(errors, s.MQOptions.Validate()...)
 	errors = append(errors, s.LogOptions.Validate()...)
 	errors = append(errors, s.AuthenticationOptions.Validate()...)
+	if err = s.AuditOptions.Validate(); err != nil {
```
Comment (x893675) [inline cmd/kubeclipper-server/app/options/options.go:69]: rewrite to  append(errors, s.AuditOptions.Validate())

**Review by lixd — APPROVED** [review]
/lgtm

**Pull request merged**